docs/API.md

```diff
@@ -201,6 +201,19 @@ Go back one entry in the history.
 ##### `goForward()`
 Go forward one entry in the history.
 
+##### `setRouteLeaveHook(route, hook)`
+Registers the given hook function to run before leaving the given route.
+
+During a normal transition, the hook function receives the next location
+as its only argument. Returning false will abort the transition. Any other return values will not have any effect.
```

Comment:
Still not quite right – you can return

1) A string to show a prompt
2) `false` to cancel the transition
3) Anything else (or nothing) to let the transition continue